unified-runtime/test/conformance/testing/include/uur/fixtures.h

```diff
@@ -396,8 +396,9 @@ struct urHostPipeTest : urQueueTest {
 
     UUR_RETURN_ON_FATAL_FAILURE(uur::KernelsEnvironment::instance->LoadSource(
         "foo", platform, il_binary));
-    ASSERT_SUCCESS(uur::KernelsEnvironment::instance->CreateProgram(
-        platform, context, device, *il_binary, nullptr, &program));
+    UUR_ASSERT_SUCCESS_OR_UNSUPPORTED(
```

Comment:
I'd prefer a KNOWN_FAILURE_ON like we have for native_cpu [here](https://github.com/intel/llvm/pull/17747/files#diff-8950f01a20bb253d30df90714d975474b4e878b2781582f95bc7197196b6c375R371), compiler support isn't optional so we should explicitly track where it's missing (and be informed when new backends don't support it)

Reply:
Revamped this logic.